wakuv2/filter_manager.go

```diff
@@ -0,0 +1,360 @@
+						}
+					}
+					mgr.resubscribe(filterID)
+				}
+
+			case FilterEventSubscribeResult:
+				subs, found := mgr.filterSubs[ev.filterID]
+				if ev.result {
+					if found {
+						subs[ev.sub.ID] = ev.sub
+						go mgr.runFilterSubscriptionLoop(ev.sub)
+					} else {
+						// We subscribed to a filter that is already uninstalled; invoke unsubscribe
+						go mgr.unsubscribeFromFilter(ev.filterID, ev.sub)
+					}
+				}
+				if found {
+					// Delete temp subscription record
+					delete(subs, ev.tempID)
+				}
+
+			case FilterEventUnsubscribeResult:
+				mgr.logger.Debug("FILTER event UNSUBSCRIBE_RESULT", zap.String("filterId", ev.filterID), zap.Stringer("peerID", ev.sub.PeerID))
+
+			case FilterEventGetStats:
+				stats := make(FilterSubs)
+				for id, subs := range mgr.filterSubs {
+					stats[id] = make(subscription.SubscriptionSet)
+					for subID, sub := range subs {
+						if sub == nil {
+							// Skip temp subs
+							continue
+						}
+
+						stats[id][subID] = sub
+					}
+				}
+				ev.ch <- stats
+			}
+		}
+	}
+}
+
+func (mgr *FilterManager) subscribeToFilter(filterID string, peer peer.ID, tempID string) {
+
+	f := mgr.getFilter(filterID)
+	if f == nil {
+		mgr.logger.Error("FILTER subscribeToFilter: No filter found", zap.String("id", filterID))
+		mgr.eventChan <- FilterEvent{eventType: FilterEventSubscribeResult, filterID: filterID, tempID: tempID, result: false}
+		return
+	}
+	contentFilter := mgr.buildContentFilter(f.PubsubTopic, f.ContentTopics)
+	mgr.logger.Debug("FILTER subscribe to filter node", zap.Stringer("peer", peer), zap.String("pubsubTopic", contentFilter.PubsubTopic), zap.Strings("contentTopics", contentFilter.ContentTopicsList()))
+	ctx, cancel := context.WithTimeout(mgr.ctx, requestTimeout)
+	defer cancel()
+
+	subDetails, err := mgr.node.FilterLightnode().Subscribe(ctx, contentFilter, filter.WithPeer(peer))
+	success := err == nil
+	var sub *subscription.SubscriptionDetails
+	if err != nil {
+		mgr.logger.Warn("FILTER could not add wakuv2 filter for peer", zap.String("filterId", filterID), zap.Stringer("peer", peer), zap.Error(err))
+	} else {
+		mgr.logger.Debug("FILTER subscription success", zap.String("filterId", filterID), zap.Stringer("peer", peer), zap.String("pubsubTopic", contentFilter.PubsubTopic), zap.Strings("contentTopics", contentFilter.ContentTopicsList()))
+		sub = subDetails[0]
+
+	}
+
+	mgr.eventChan <- FilterEvent{eventType: FilterEventSubscribeResult, filterID: filterID, tempID: tempID, sub: sub, result: success}
+}
+
+func (mgr *FilterManager) unsubscribeFromFilter(filterID string, sub *subscription.SubscriptionDetails) {
+	mgr.logger.Debug("FILTER unsubscribe from filter node", zap.String("filterId", filterID), zap.String("subId", sub.ID), zap.Stringer("peer", sub.PeerID))
+	// Unsubscribe on light node
+	ctx, cancel := context.WithTimeout(mgr.ctx, requestTimeout)
+	defer cancel()
+	_, err := mgr.node.FilterLightnode().UnsubscribeWithSubscription(ctx, sub)
+	success := err == nil
```

Comment:
I would move this line below the error checking, it always looks a bit odd using a value before error checking, though the code is perfectly fine

Reply:
Moved.